[BUG] Publish command fails with extra switches

**Describe the bug**
Running a target that uses the publish executor seems to fail with any additional switches

Seems to be introduced by this change in this commit:
https://github.com/nx-dotnet/nx-dotnet/commit/fa8635518a5cd6cb0c262dc8926b8d32bfa894ca

**To Reproduce**
Steps to reproduce the behaviour:
In my repo I run the following command but in my testing, it seems to occur with any command that has additional switches
`npx nx run-many --target=deploy --all --prod --extraParameters="--self-contained=false /p:CopyOutputSymbolsToPublishDirectory=false /p:Version=2022.03.25.1 /p:VersionAssembly=2022.03.25.1 "`

eg. When I add --prod, it fails as well.

1. Run a command that uses the publish executor. eg `nx run project-name:deploy --prod`
2. See error in the build output `MSBUILD : error MSB1008: Only one project can be specified`

**Expected behaviour**
The command executes without error, and if there is an MSBuild error, the task should fail instead of reporting that the task executed successfully

**Screenshots**
![image](https://user-images.githubusercontent.com/10575019/160121315-3962623e-3ae6-4378-a7af-746cb6586620.png)

**Environment:**

- Browser [e.g. chrome, safari]
- Version 1.9.6

**Additional context**
I have the `deploy` target pointing to the `@nx-dotnet/core:publish` executor

--- a/packages/dotnet/.babelrc
+++ b/packages/dotnet/.babelrc
@@ -1,3 +1,0 @@
-{
-  "presets": [["@nrwl/web/babel", { "useBuiltIns": "usage" }]]
-}

--- a/packages/dotnet/src/lib/core/dotnet.client.spec.ts
+++ b/packages/dotnet/src/lib/core/dotnet.client.spec.ts
@@ -1,0 +1,68 @@
+import { DotNetClient } from './dotnet.client';
+import { mockDotnetFactory } from './dotnet.factory';
+import * as cp from 'child_process';
+
+describe('dotnet client', () => {
+  describe('publish', () => {
+    describe('extra parameters', () => {
+      const dotnetClient = new DotNetClient(mockDotnetFactory());
+
+      let spawnSyncSpy: jest.SpyInstance;
+
+      beforeEach(() => {
+        spawnSyncSpy = jest
+          .spyOn(cp, 'spawnSync')
+          .mockImplementation(jest.fn());
+      });
+
+      afterEach(() => {
+        jest.resetAllMocks();
+      });
+
+      it('should handle multiple parameters', () => {
+        dotnetClient.publish(
+          'my-project',
+          undefined,
+          undefined,
+          '--flag --other-flag',
+        );
+        expect(spawnSyncSpy.mock.calls[0][1]).toContain('--flag');
+        expect(spawnSyncSpy.mock.calls[0][1]).toContain('--other-flag');
+        expect(spawnSyncSpy).toHaveBeenCalledTimes(1);
+      });
+
+      it('should handle multiple parameters with quotations', () => {
+        dotnetClient.publish(
+          'my-project',
+          undefined,
+          undefined,
+          `--flag \\p:"my project"`,
+        );
+        expect(spawnSyncSpy.mock.calls[0][1]).toContain('--flag');
+        expect(spawnSyncSpy.mock.calls[0][1]).toContain('\\p:"my project"');
+        expect(spawnSyncSpy).toHaveBeenCalledTimes(1);
+      });
+
+      it('should handle several parameters', () => {
+        dotnetClient.publish(
+          'my-project',
+          undefined,
+          undefined,
+          `--self-contained=false /p:CopyOutputSymbolsToPublishDirectory=false /p:Version=2022.03.25.1 /p:VersionAssembly=2022.03.25.1 /p:Name:"My Project"`,
+        );
+        expect(spawnSyncSpy.mock.calls[0][1]).toMatchInlineSnapshot(`
+          Array [
+            "publish",
+            "\\"my-project\\"",
+            "--self-contained=false",
+            "/p:CopyOutputSymbolsToPublishDirectory=false",
+            "/p:Version=2022.03.25.1",
+            "/p:VersionAssembly=2022.03.25.1",
+            "/p:Name:\\"My Project\\"",
+          ]
+        `);
+        expect(spawnSyncSpy).toHaveBeenCalledTimes(1);
+      });
+    });
+  });
+});

--- a/packages/dotnet/src/lib/core/dotnet.client.ts
+++ b/packages/dotnet/src/lib/core/dotnet.client.ts
@@ -110,7 +110,8 @@ export class DotNetClient {
       params.push(`-p:PublishProfile=${publishProfile}`);
     }
     if (extraParameters) {
-      params.push(`${extraParameters}`);
+      const matches = extraParameters.match(EXTRA_PARAMS_REGEX);
+      params.push(...(matches as RegExpMatchArray));
     }
     return this.logAndExecute(params);
   }
@@ -195,3 +196,10 @@ export class DotNetClient {
     });
   }
 }
+
+/**
+ * Regular Expression for Parsing Extra Params before sending to spawn / exec
+ * First part of expression matches parameters such as --flag="my answer"
+ * Second part of expression matches parameters such as --flag=my_answer
+ */
+const EXTRA_PARAMS_REGEX = /\S*".+?"|\S+/g;

